### PR TITLE
Autosuggest

### PR DESCRIPTION
**Pull request by RomeshA**
Add function to retrieve suggestions based on Levenshtein distance. How the user should be informed of suggestions is highly context dependent - therefore this function is primarily aimed at returning suggestions for the calling code to format and present as appropriate. The `die` argument results in a standard 'Did you mean' error being raised, for cases where sophisticated formatting and error handling is not required

### CODE REVIEW

File: sciris/sc_utils.py [hunk 1]
```diff
@@ -1553,8 +1554,61 @@ def flattendict(inputdict=None, basekey=None, subkeys=None, complist=None, keyli
...
+    'Foo'
+    >>> suggest('foo',['Foo ','boo'])
+    'Foo '
 
+    """
+
+    if isstring(valid_inputs):
+        valid_inputs = [valid_inputs]
+    else:
+        valid_inputs = list(valid_inputs)
```
Comment (cliffckerr) [inline sciris/sc_utils.py]: Could do `promotetolist()` :)

File: sciris/sc_utils.py [hunk 2]
```diff
@@ -1553,8 +1554,61 @@ def flattendict(inputdict=None, basekey=None, subkeys=None, complist=None, keyli
...
+
+    if min(distance) > threshold:
+        if die:
+            raise Exception('"%s" not found' % user_input)
+        else:
+            return None
+    elif die:
+        raise Exception('"%s" not found - did you mean "%s"?' % (user_input, suggestion))
+    else:
+        return suggestion
```
Comment (cliffckerr) [inline sciris/sc_utils.py]: Would also suggest an optional kwarg `fulloutput=False` (or a better name!), which if `True` returns an odict of `{valid_input[i]:distance[i]}` pairs

File: sciris/sc_utils.py (hunk 1 above)
Comment (RomeshA) [reply sciris/sc_utils.py]: Did that first, but it fails if you pass in a set because it just wraps the set in a list!

Comment (cliffckerr) [reply sciris/sc_utils.py]: That sounds like a bug in `promotetolist()` that should be fixed :)

**Comment by RomeshA** [comment]
Couple of other issues before merging

- The error handling `raise Exception(errormsg).with_traceback(E.__traceback__)` doesn't seem to be supported by Python 2? The tox test is failing with `'exceptions.Exception' object has no attribute 'with_traceback'`. For Python 3, it might be easier to just do `raise Exception(errormsg) from E` and then do something different for Python 2 - could even consider not bothering with the traceback?
- Type validation for `promotetolist` is a little broken if the first argument isn't iterable e.g. `sc.promotetolist(1,str)` fails with `'int' object is not iterable` whereas `sc.promotetolist('1',int)` fails with `object is <class 'str'>, but <class 'int'> is required`. On line 955 it probably should explicitly check whether `obj` is iterable and wrap it in a list first
- Type validation gets skipped if the container is a list e.g.
```
sc.promotetolist({1,2,3},str) -> error (expected)
sc.promotetolist([1,2,3],str) -> [1, 2, 3] (succeeds, probably unexpectedly)
```

**Pull request merged**